tests/integration_test/dynamic_routing_test/dynamic_routing_integration_test.go

```diff
@@ -525,22 +551,38 @@ func TestDynamicBackendRoutingTLS(t *testing.T) {
 			},
 		},
 	}
+
 	for _, tc := range testData {
-		url := fmt.Sprintf("http://localhost:%v%v", s.Ports().ListenerPort, tc.path)
-		// var gotResp []byte
-		var err error
-		_, err = client.DoPost(url, tc.message)
+		func() {
+			s := env.NewTestEnv(comp.TestDynamicBackendRoutingTLS, platform.EchoRemote)
+			s.UseWrongBackendCertForDR(tc.useWrongCert)
+			defer s.TearDown()
 
-		if err == nil || !strings.Contains(err.Error(), tc.wantError) {
-			t.Errorf("Test (%s): failed, want error: %v, got error: %v", tc.desc, tc.wantError, err)
-		}
+			if err := s.Setup(testDynamicRoutingArgs); err != nil {
+				t.Fatalf("fail to setup test env, %v", err)
```

Comment:
change to "Failf"  or "Errorf"?

Reply:
Based on go readiability guidance:

"t.Fatal is usually only appropriate when some piece of test setup fails, without which you cannot run the test at all. "